stable/mongodb-replicaset/tests/mongodb-up-test.sh

```diff
@@ -1,12 +1,43 @@
 #!/usr/bin/env bash
 
-MONGOCACRT=/ca/tls.crt
+CACRT_FILE=/work-dir/tls.crt
+CAKEY_FILE=/work-dir/tls.key
 MONGOPEM=/work-dir/mongo.pem
 
 MONGOARGS="--quiet"
 
-if [ -f "$MONGOPEM" ]; then
-    MONGOARGS="$MONGOARGS --ssl --sslCAFile $MONGOCACRT --sslPEMKeyFile $MONGOPEM"
+if [ -n "${CACRT}" ]; then
+    log "Generating certificate"
+    echo "${CACRT}" > ${CACRT_FILE}
+    echo "${CAKEY}" > ${CAKEY_FILE}
+
+    # Move into /work-dir
+    pushd /work-dir
+
+cat >openssl.cnf <<EOL
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = $(echo -n "$(hostname)" | sed s/-[0-9]*$//)
+DNS.2 = $(hostname)
```

Comment:
The test pod has a different hostname from the mongod process and this will cause the certificate to not match the server host. This needs to be either the pod hostname (see `pod_name()`) or the service name. 

Alternatively, you could potentially just skip the pem generation and just pass `--ssl --sslCAFile $CACRT_FILE --sslAllowInvalidCertificates`. You won't need a workdir with this approach.

With either approach, just mount the ca secret as a volume.